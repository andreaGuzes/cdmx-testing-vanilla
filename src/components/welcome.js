/* eslint-disable quotes */
/* eslint-disable no-param-reassign */

export async function bienvenida(target, firebase) {
  const html = `
    <h1>Bienvenida a mi Red social</h1>
    <button id="botoncito">Soy un boton</button>
    <div id="postContainer"></div>
  `;

  target.innerHTML = html;
  const posts = await firebase.getPosts();
  const container = document.getElementById('postContainer');
  const templates = posts.map((post) => `<p>${post.message}</p>`);
  container.innerHTML = templates.join('');
  const boton = document.getElementById('botoncito');
  boton.addEventListener('click', () => {
    const imprimir = document.createElement('p');
    imprimir.textContent = 'Ya llegue';
    container.appendChild(imprimir);
  });
}
