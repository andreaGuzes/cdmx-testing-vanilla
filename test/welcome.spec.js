/* eslint-disable quotes */
import { bienvenida } from '../src/components/welcome.js';

describe('new suite', () => {
  beforeEach(() => {
    document.body.innerHTML = '<div id="app"></div>';
  });

  test('deberia renderizar', async () => {
    const target = document.getElementById('app');
    const posts = [
      {
        message: 'test',
      },
    ];
    const getPosts = jest.fn().mockImplementation(() => Promise.resolve(posts));
    const firebase = { getPosts };

    await bienvenida(target, firebase);

    expect(target.innerHTML).toMatchSnapshot();
  });

  test('boton cambia texto', async () => {
    const target = document.getElementById('app');
    const getPosts = jest.fn().mockImplementation(() => Promise.resolve([]));
    const firebase = { getPosts };

    await bienvenida(target, firebase);

    const botoncito = document.getElementById('botoncito');
    botoncito.click();

    expect(target.innerHTML).toMatchSnapshot();
  });
});
