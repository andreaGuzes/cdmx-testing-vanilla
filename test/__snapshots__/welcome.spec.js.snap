// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new suite boton cambia texto 1`] = `
"
    <h1>Bienvenida a mi Red social</h1>
    <button id=\\"botoncito\\">Soy un boton</button>
    <div id=\\"postContainer\\"><p>Ya llegue</p></div>
  "
`;

exports[`new suite deberia renderizar 1`] = `
"
    <h1>Bienvenida a mi Red social</h1>
    <button id=\\"botoncito\\">Soy un boton</button>
    <div id=\\"postContainer\\"><p>test</p></div>
  "
`;
